crates/rpc/rpc/src/otterscan.rs

```diff
@@ -86,11 +88,55 @@ where
     /// Handler for `getBlockTransactions`
     async fn get_block_transactions(
         &self,
-        _block_number: BlockNumberOrTag,
-        _page_number: usize,
-        _page_size: usize,
+        block_number: BlockNumberOrTag,
+        page_number: usize,
+        page_size: usize,
     ) -> RpcResult<OtsBlockTransactions> {
-        Err(internal_rpc_err("unimplemented"))
+        // retrieve full block and its receipts
+        let block = self.eth.block_by_number(block_number, true);
+        let receipts = self.eth.block_receipts(BlockId::Number(block_number));
+        let (block, receipts) = futures::try_join!(block, receipts)?;
+
+        let block = block.ok_or_else(|| internal_rpc_err("block not found"))?;
+        let mut receipts = receipts.ok_or_else(|| internal_rpc_err("receipts not found"))?;
+
+        // check if the number of transactions matches the number of receipts
+        let tx_len = block.transactions.len();
+        if tx_len != receipts.len() {
+            return Err(internal_rpc_err(
+                "the number of transactions does not match the number of receipts",
+            ));
+        }
+
+        // make sure the block is full
+        let BlockTransactions::Full(transactions) = &block.transactions else {
+            return Err(internal_rpc_err("block is not full"));
+        };
+
+        // Crop page
+        let page_end = tx_len.saturating_sub(page_number * page_size);
+        let page_start = page_end.saturating_sub(page_size);
+
+        // Crop transactions
+        let mut transactions = transactions[page_start..page_end].to_vec();
```

Comment:
replace with .truncate() + .drain to retain the range

Reply:
Drains should be enough?